partner_firstname/__manifest__.py

```diff
@@ -18,7 +18,7 @@
     "maintainer": "Camptocamp, Acsone",
     "category": "Extra Tools",
     "website": "https://github.com/OCA/partner-contact",
-    "depends": ["base_setup"],
+    "depends": ["base_setup", "partner_type_base"],
```

Comment:
doing it in a current version could lead to issues for database already using partner_firstname as the dependency won't install itself after pulling this new version and updating the database. People updating the code repository are not always aware of those changes. And debugging it could lead to loss of time and frustration. 

Maybe setting partner_type_base as auto_install True could avoid the issue.

what do you think ?

Reply:
I was note aware of that. I thought if we change the version number, also the dependencies will be rechecked and installed. 

auto_install would be possible, but this also leads to unnecessary install databases which doesn't need it, correct?

I'll test it first, but an upgrade script might be the solution?

Reply:
yes auto_install True will install it on all databases. upgrade script could be the solution.

Reply:
to be precise this is the update all module that will lead to the issue. updating partner_firstname should install the module indeed. But you need to be aware of such change in the dependency.

the issue was partially fixed with the PR linked to this issue https://github.com/odoo/odoo/issues/72703 and merged on v14. It seems not having evolved in 16.0. I don't know for 17.0 and 18.0.

Reply:
As already said, you shouldn't change dependencies on a stable module that is already merged in this version.

Reply:
@houssine78 thank you for the link. Very interesting.

@pedrobaeza I understand your point. The thing is, that a lot of modules are migrate 1:1 very fast under a lot of time pressure and these kind of changes will never be done/time for. 

The migration to v18 is already completed, so the next possible time is the migration to v19. So I need to develop now a feature, which we can start use in 2 years? I don't think this is working eighter. 

I totally agree, that this kind of change should be done by odoo in odoo directly with a version change, but I don't think this will ever be applied.

If this PR will not be merged, it is like it is. But I apricate your input and I'll fix the found issues soon.

Reply:
You can do this change in v18, but not on a version which is already one year old.

Reply:
OK, I'll prepare a new PR for v18 and make backport for myself. I can live with this very well.